### ISSUE TITLE
Creating a Multiboot USB Drive

### ISSUE THREAD

**Issue by TheCyberSystem** [ventoy/Ventoy#869]
I bought the SanDisk Ultra 128GB Dual Drive (SDDDC2-128G-G46) with the hopes of turning it into a multiboot drive that I could use on a computer whether it has USB-A or USB-C. If I can get it to work, I'm wanting to have installation media for Ubuntu and Windows 10, and then another partition to store program installers on.

I've had no end of trouble with this so far. Originally I tried using WinSetupFromUSB which didn't go well at all. I'm now trying Ventoy which seems to get further in the process, but the program fails to partition/boot/format the USB drive, which is weird because it works fine on another 8GB stick I have. I asked SanDisk and they said I should ask Microsoft. They stated it was not their hardware that was the problem, but rather something in Windows.

> Also, it appears that you are facing the issue during creating the boot. In this case, we request you to contact Windows support to know more about FAT32 and boot option.
> 
> Note - SanDisk do not have any expertise for Boot enabled interface and its creation/ issue involved with it.

Ventoy seems to prefer formatting drives as exFAT (NTFS can be used in a pinch though it can cause Ventoy some issues).

Here's the highlights of the program log:

> [2021/04/29 19:50:01.261] Deleting all partitions ......................... OK
> [2021/04/29 19:50:01.270] Lock disk for write .............................
> [2021/04/29 19:50:01.275] QueryDosDeviceA success \Device\Harddisk1\DR5
> [2021/04/29 19:50:01.303] [0] CreateFileA \\.\PhysicalDrive1 code:0 00000358
> [2021/04/29 19:50:01.304] Opened \\.\PhysicalDrive1 for exclusive write access
> [2021/04/29 19:50:01.313] FSCTL_LOCK_VOLUME success
> [2021/04/29 19:50:01.320] Wait for format part1 ...
> [2021/04/29 19:50:06.335] Formatting part1 exFAT ...
> [2021/04/29 19:50:06.336] Formatting Part1 exFAT ...
Then a bunch of WriteFile errors before
> [2021/04/29 19:50:06.564] Formatting Part1 exFAT success
> [2021/04/29 19:50:06.570] mount part 13
> [2021/04/29 19:50:06.575] mount exfat failed 13
> [2021/04/29 19:50:06.581] FormatPart1exFAT failed.
> [2021/04/29 19:50:06.588] FindProcessOccupyDisk for PhyDrive 1
> [2021/04/29 19:50:06.593] QueryDosDeviceA success \Device\Harddisk1\DR5
> [2021/04/29 19:50:06.728] NOTE: Could not identify the process(es) or service(s) accessing \Device\Harddisk1\DR5
> [2021/04/29 19:50:08.311] Not ventoy disk, clear secure boot option

I also found out that Windows Antivirus had blocked Ventoy from accessing "restricted memory" and added an exception. However I reran Ventoy and it still had the same error.

**Comment by maxmod** [comment]
format the usb before using ventoy use windows format tool or linux gparted or formatusb from https://formatusb.com before ventoy

**Comment by TheCyberSystem** [comment]
Not going to lie, I'm legitimately amazed that worked first try. Ran the "FormatUSB" tool with no errors. Ran Ventoy, and it worked straight away. I'm so pleased that I finally got it working - I was considering throwing out the drive because I thought it was some stupid hardware thing with the drive itself, maybe some kind of firmware chip that allowed it to switch between the USB-A and USB-C connectors.

Thanks maxmod for the tip!
